OPHD/Technology/ResearchTracker.cpp

```diff
@@ -0,0 +1,38 @@
+#include "ResearchTracker.h"
+
+#include <algorithm>
+#include <iterator>
+#include <stdexcept>
+
+
+void ResearchTracker::startResearch(int techId, int progress, int assigned)
+{
+	if (std::find(mCompleted.begin(), mCompleted.end(), techId) != mCompleted.end())
+	{
+		throw std::runtime_error("Can't update a technology that's already researched.");
+	}
+
+	auto emplacedItem = mTechnologiesBeingResearched.emplace(techId, std::make_tuple(progress, assigned));
+	if (!emplacedItem.second)
+	{
+		(*emplacedItem.first).second = {progress, assigned};
+	}

```

Comment:
I just noticed this. You can use [`std::map::insert_or_assign`](https://en.cppreference.com/w/cpp/container/map/insert_or_assign) instead of [`std::map::emplace`](https://en.cppreference.com/w/cpp/container/map/emplace). That would eliminate the need for the `if` block to manually write the data when the key already exists.

It was tempting to use `operator[]`, though that would default construct an object before assignment when the key isn't found. Using `insert_or_assign` avoid the needless default construction.